### PR TITLE
WT-10152 Add a home directory option to the cppsuite

### PR DESCRIPTION
**Pull request by luke-pearson**

### CODE REVIEW

File: test/cppsuite/tests/run.cpp [hunk 1]
```diff
@@ -88,6 +88,7 @@ print_help()
     std::cout << std::endl;
     std::cout << "SYNOPSIS" << std::endl;
     std::cout << "\trun [OPTIONS]" << std::endl;
+    std::cout << "\trun -H [HOME]" << std::endl;
```
Comment (etienneptl) [inline test/cppsuite/tests/run.cpp:91]: Can we A->Z please

File: test/cppsuite/tests/run.cpp [hunk 2]
```diff
@@ -108,6 +109,7 @@ print_help()
     std::cout << std::endl;
     std::cout << "OPTIONS" << std::endl;
     std::cout << "\t-h Output a usage message and exit." << std::endl;
+    std::cout << "\t-H Specifies the database home directory." << std::endl;
```
Comment (etienneptl) [inline test/cppsuite/tests/run.cpp:112]: Same, A->Z, btw, the `-h` should probably be before the `-f` option.

File: test/cppsuite/tests/run.cpp [hunk 3]
```diff
@@ -234,6 +240,11 @@ main(int argc, char *argv[])
             for (const auto &test_name : all_tests)
                 std::cout << "\t" << test_name << std::endl;
             return 0;
+        } else if (option == "-H") {
```
Comment (etienneptl) [inline test/cppsuite/tests/run.cpp:243]: Wish we could have A->Z in this function too but might be too much work to re sort everything, up to you.

**Review by etienneptl — APPROVED** [review]
LGTM, only nit picks, I assume you tested different home dirs!

**Review by clarissecheah — APPROVED** [review]
LGTM

File: test/cppsuite/tests/run.cpp (hunk 1 above)
Comment (luke-pearson) [reply test/cppsuite/tests/run.cpp]: [eb6d460](https://github.com/wiredtiger/wiredtiger/pull/8484/commits/eb6d460cf9f91e632783d4cbe5537064096d2264)

File: test/cppsuite/tests/run.cpp (hunk 2 above)
Comment (luke-pearson) [reply test/cppsuite/tests/run.cpp]: [eb6d460](https://github.com/wiredtiger/wiredtiger/pull/8484/commits/eb6d460cf9f91e632783d4cbe5537064096d2264)

File: test/cppsuite/tests/run.cpp (hunk 3 above)
Comment (luke-pearson) [reply test/cppsuite/tests/run.cpp:243]: Nah not in this change, theres a bunch of option cleanup work we could do which I think should be separate. Such as removing all the local variables and just using the test_args structure.

**Review by etienneptl — APPROVED** [review]

**Comment by luke-pearson** [comment]
evergreen merge

**Pull request merged**